### PR TITLE
[Java][webclient/resttemplate] fix dependencies for gradle with jakarta enabled

### PR DESCRIPTION
**Pull request by ChristianMoesl**
This PR fixes build errors, because of wrong dependencies in the gradle build templates for java clients with spring dependencies (webclient & resttemplate) and `useJakartaEe` enabled.

This error can be reproduced quite easily on `master` branch with the following commands:
```bash
cd samples/client/petstore/java/webclient-jakarta
chmod +x gradlew && gradle clean build
```
It will result in build errors like these:
image

What was done:
- remove swagger dependency, if it isn't in use
- use jakarta >= 2.0 if `useJakartaEe` is enabled (only jakarta 2.0 and above exposes annotations in `jakarta.annotations.*` namespace)
- use spring boot 3 and spring 6 for jakarta support
- added a Gradle build step to CI

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/Java/libraries/webclient/build.gradle.mustache [hunk 1]
```diff
@@ -78,8 +84,14 @@ if(hasProperty('target') && target == 'android') {
     apply plugin: 'java'
     apply plugin: 'maven-publish'
 
+    {{#useJakartaEe}}
+    sourceCompatibility = JavaVersion.VERSION_11
```
Comment (borsch) [inline modules/openapi-generator/src/main/resources/Java/libraries/webclient/build.gradle.mustache:88]: why 11?

**Review by borsch — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/main/resources/Java/libraries/webclient/build.gradle.mustache (hunk 1 above)
Comment (ChristianMoesl) [reply modules/openapi-generator/src/main/resources/Java/libraries/webclient/build.gradle.mustache]: Was a mistake. I forgot to push my latest changes. Is now up 2 date. Thanks!

**Review by borsch — APPROVED** [review]

**Comment by borsch** [comment]
<html><body>


Java | @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala  (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01)  @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)
-- | --
Java Spring | @cachescrubber (2022/02) @welshm (2022/02) @MelleD  (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02)  @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)


</body>
</html>

**Review by welshm — APPROVED** [review]
Nice!

**Comment by borsch** [comment]
@ChristianMoesl could you please check failed pipelines? For "Samples up-to-date" just pull latest master and ensure that you have regenerated samples for all project that are affected by your change

**Comment by ChristianMoesl** [comment]
@borsch I fixed the issues, which caused the pipeline to fail. There were 3 things to fix:
- `classifier` property was removed in Gradle 8 and needed to be replaced by `archiveClassifier`
- `main` property is deprecated in Gradle 8 and needed to be replaced by `mainClass` (to future proof it)
- I had to generate all samples, not just the ones I used in my test case 😅

**Comment by borsch** [comment]
Not sure that exclusively supporting grade 8 is a good idea.. This simply means that we would force all clients of this generator to start using grade 8

FYI @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

**Comment by ChristianMoesl** [comment]
Sry, I should have clarified the Gradle compatibility for this. 
`mainClass` was introduced in [Gradle 6.7](https://docs.gradle.org/6.7/javadoc/org/gradle/api/tasks/JavaExec.html#getMainClass--)
`archiveClassifier` was introduced in [Gradle 5.1](https://docs.gradle.org/5.1/javadoc/org/gradle/api/tasks/bundling/AbstractArchiveTask.html#getArchiveClassifier())

So overall, the generated code will be compatible with Gradle >= 6.7, which sounds reasonable to me.

**Pull request merged**